internal/health/health.go

```diff
@@ -162,8 +168,9 @@ func (h *Health) IsAlive() bool {
 func (h *Health) checkAlive() bool {
 	// if any counter is 0, we're dead
 	for source, a := range h.timeLeft {
-		if a == 0 {
+		if a == 0 && h.alives[source] {
 			h.Logger.Error().WithField("source", source).Logf("IsAlive: source dead due to timeout")
+			h.alives[source] = false
```

Comment:
Should we change the mutex on line 161 to a write lock since we are changing `h.alives` here?